client/lib/analytics/ad-tracking.js

```diff
@@ -514,76 +491,105 @@ export function retarget( urlPath ) {
 
 	// Quantcast
 	if ( isQuantcastEnabled ) {
-		window._qevents.push( {
+		const params = {
 			qacct: TRACKING_IDS.quantcast,
 			event: 'refresh',
-		} );
-		debug( 'Retargeting: Quantcast' );
+		};
+		debug( 'retarget: [Quantcast]', params );
+		window._qevents.push( params );
 	}
 
 	// Facebook
 	if ( isFacebookEnabled ) {
-		window.fbq( 'trackSingle', TRACKING_IDS.facebookInit, 'PageView' );
-		debug( 'Retargeting: Facebook' );
+		const params = [ 'trackSingle', TRACKING_IDS.facebookInit, 'PageView' ];
+		debug( 'retarget: [Facebook]', params );
+		window.fbq( ...params );
 	}
 
-	// Rate limited retargeting
+	if ( isBingEnabled ) {
+		debug( 'retarget: [Bing]' );
+		window.uetq.push( 'pageLoad' );
+	}
 
-	const nowTimestamp = Date.now() / 1000;
-	if ( nowTimestamp < lastRetargetTime + retargetingPeriod ) {
-		return;
+	// Wordpress.com Google Ads Gtag
+	if ( isWpcomGoogleAdsGtagEnabled ) {
+		const params = [ 'config', TRACKING_IDS.wpcomGoogleAdsGtag, { page_path: urlPath } ];
+		debug( 'retarget: [Google Ads] WPCom', params );
+		window.gtag( ...params );
 	}
-	lastRetargetTime = nowTimestamp;
 
-	// Twitter
-	if ( isTwitterEnabled ) {
-		window.twq( 'track', 'PageView' );
-		debug( 'Retargeting: [rate limited] Twitter' );
+	// Jetpack Google Ads Gtag
+	if ( isJetpackGoogleAdsGtagEnabled ) {
+		const params = [ 'config', TRACKING_IDS.jetpackGoogleAdsGtag, { page_path: urlPath } ];
+		debug( 'retarget: [Google Ads] Jetpack', params );
+		window.gtag( ...params );
 	}
 
 	// AdWords
 	if ( isAdwordsEnabled ) {
+		const params = {
+			google_conversion_id: ADWORDS_CONVERSION_ID,
+			google_remarketing_only: true,
+		};
+		debug( 'retarget: [AdWords]', params );
 		if ( window.google_trackConversion ) {
-			window.google_trackConversion( {
-				google_conversion_id: ADWORDS_CONVERSION_ID,
-				google_remarketing_only: true,
-			} );
+			window.google_trackConversion( params );
 		}
-		debug( 'Retargeting: [rate limited] AdWords' );
 	}
 
 	// Floodlight
 	recordPageViewInFloodlight( urlPath );
 
-	// Yahoo Gemini
-	if ( isGeminiEnabled ) {
-		new Image().src = YAHOO_GEMINI_AUDIENCE_BUILDING_PIXEL_URL;
-		debug( 'Retargeting: [rate limited] Yahoo' );
+	// Pinterest
+	if ( isPinterestEnabled ) {
+		debug( 'retarget: [Pinterest]' );
+		window.pintrk( 'page' );
 	}
 
-	// Quora
-	if ( isQuoraEnabled ) {
-		window.qp( 'track', 'ViewContent' );
-		debug( 'Retargeting: [rate limited] Quora' );
-	}
+	// Rate limited retargeting
```

Comment:
Yes it's intentional. Actually, I wouldn't mind phasing out this rate-limiting logic or at least use it less aggressively, which was introduced probably mostly to appease the worries about tracking being too intrusive than anything else for which I'm not sure we've had all that evidence at the cost of the reliability of our trackers.

The idea I followed is to simply have our main trackers work normally and keep the rest rate limited, plus Pinterest which is a new tracker, in order to give it the best chance of success.

For Gtag tags in particular since they use the data layer the rate limiting doesn't make much sense, and since Gtag is non rate limited it made sense to also have the AdWords non rate limited so we can compare apples to apples if creating audiences from the two.

One thing we could do is to go through each tracker and ask our fellow marketers which ones they would like to have rate-limiting on or off. If people are not actively creating audiences based on Calypso pages then it makes sense to rate limit. However in reality today we could get a response in one way and 6 months from now somebody might decide to make an experiment based on the last 6 months of data of some specific page patterns, data which we would not have for rate limited trackers. So there is an opportunity cost there.  Just something to keep in mind.

cc @jaswrks for thoughts on the subject.